sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1521,6 +1521,40 @@ case class DateAddYMInterval(date: Expression, interval: Expression) extends Add
   override def sql: String = s"${left.sql} + ${right.sql}"
 }
 
+// Adds the year-month interval to the timestamp
+case class TimestampAddYMInterval(
```

Comment:
Shall we make it general and accept both year-month and day-time intervals? The implementation can have 2 branches:
```
override def nullSafeEval(micros: Any, months: Any): Any = interval.dataType match {
  case YearMonthIntervalType =>
    timestampAddMonths(micros.asInstanceOf[Long], months.asInstanceOf[Int], zoneId)
  case DayTimeIntervalType => ...
}
```

Reply:
I plan to support `timestamp + day-time interval` separately in another PR. The implementation itself is not complex, need to write more tests.